src/bbs-plus/signature.ts

```diff
@@ -26,7 +28,7 @@ export abstract class Signature extends BytearrayWrapper {
    * @param message
    */
   static encodeMessageForSigning(message: Uint8Array): Uint8Array {
```

Comment:
Since the encoding logic is going to remain same for each sig scheme, better to move this and 3 functions below to an interface like `SigningMessageEncoder`